include/engine/guidance/assemble_geometry.hpp

```diff
@@ -78,11 +78,11 @@ inline LegGeometry assembleGeometry(const datafacade::BaseDataFacade &facade,
         }
 
         prev_coordinate = coordinate;
-        geometry.annotations.emplace_back(
-            LegGeometry::Annotation{current_distance,
-                                    path_point.duration_until_turn / 10.,
-                                    path_point.weight_until_turn / facade.GetWeightMultiplier(),
-                                    path_point.datasource_id});
+        geometry.annotations.emplace_back(LegGeometry::Annotation{
+            current_distance,
+            (path_point.duration_until_turn - path_point.duration_of_turn) / 10.,
```

Comment:
For a major change this is a bit hidden - can we make this more prominent and document why we need to do this here